packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1602,11 +1602,52 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
   ///  * [Form], which provides an `onWillPop` callback that uses this mechanism.
   ///  * [addScopedWillPopCallback], which adds callback to the list
   ///    checked by [willPop].
+  @Deprecated(
+    'Use registerPopScope or PopScope instead. '
+    'This feature was deprecated after v3.12.0-1.0.pre.',
+  )
   void removeScopedWillPopCallback(WillPopCallback callback) {
     assert(_scopeKey.currentState != null, 'Tried to remove a willPop callback from a route that is not currently in the tree.');
     _willPopCallbacks.remove(callback);
   }
 
+  /// Registers the existence of a [PopScope] widget in the route.
+  ///
+  /// [PopScope] widgets registered in this way will have their
+  /// [PopScope.onPopped] callbacks called when a route is popped or a pop is
+  /// attempted. They will also be able to block pop operations with
+  /// [PopScope.popEnabled] through this route's [popEnabled] method.
+  ///
+  /// See also:
+  ///
+  /// [unregisterPopScope], which performs the opposite operation.
+  void registerPopScope(PopScope widget) {
```

Comment:
It is kinda strange that this takes a widget, it doesn't seem to need that. Should we maybe define an intermediary interface?

Reply:
Good point, I've created PopInterface for this purpose.